controller.go

```diff
@@ -224,9 +224,9 @@ type controller struct {
 
 // MiddlewareController will return the speakeasy middleware controller from the current request,
 // if the current request is monitored by the speakeasy middleware.
-func MiddlewareController(r *http.Request) *controller {
-	c, _ := r.Context().Value(controllerKey).(*controller)
-	return c
+func MiddlewareController(r *http.Request) (*controller, bool) {
+	c, ok := r.Context().Value(controllerKey).(*controller)
```

Comment:
actually I do remember why I didn't deal with the ok variable in the end.

So it is entirely possible that c is still `nil` if ok is `true` and so the end user would need to do `ok && c != nil` so I thought I would just simplify it to `c != nil`

Reply:
ok cool, thanks for that